src/plugins/autoflow/AutoflowTabularController.js

```diff
@@ -0,0 +1,160 @@
+        row.classes = evaluate(datum).map(function (violation) {
+            return violation.cssClass;
+        }).join(" ");
+
+        this.data.updated = formatDomain(datum);
+    };
+
+    AutoflowTabularController.prototype.makeRow = function (childObject) {
+        var id = [
+            childObject.identifier.namespace,
+            childObject.identifier.key
+        ].join(":");
+
+        if (this.rows[id]) {
+            return this.rows[id];
+        }
+
+        var row = {
+            classes: "",
+            name: childObject.name,
+            value: undefined
+        };
+        var metadata = this.openmct.telemetry.getMetadata(childObject);
+        var ranges = metadata.valuesForHints(['range']);
+        var domains = metadata.valuesForHints(['domain']);
+        var rangeFormatter =
+            this.openmct.telemetry.getValueFormatter(ranges[0]);
+        var domainFormatter =
+            this.openmct.telemetry.getValueFormatter(domains[0]);
+        var evaluator =
+            this.openmct.telemetry.limitEvaluator(childObject);
+
+        this.unlistens.push(this.openmct.telemetry.subscribe(
+            childObject,
+            this.updateRow.bind(
+                this,
+                row,
+                rangeFormatter.format.bind(rangeFormatter),
+                domainFormatter.format.bind(domainFormatter),
+                function (datum) {
+                    var violations = evaluator.evaluate(datum, ranges[0]);
+                    if (!violations) {
+                        return [];
+                    }
+                    return Array.isArray(violations) ? violations : [violations];
+                }
+            )
+        ));
+
+        this.rows[id] = row;
+
+        return row;
+    };
+
+    AutoflowTabularController.prototype.update = function () {
+        var column = [];
+        var index = 0;
+        var filteredObjects =
+            this.childObjects.filter(this.matchesFilter.bind(this));
+
+        this.data.columns = [];
+
+        while (index < filteredObjects.length) {
+            if (column.length >= this.rowCount) {
+                this.data.columns.push(column);
+                column = [];
+            }
+
+            column.push(this.makeRow(filteredObjects[index]));
+            index += 1;
+        }
+
+        if (column.length > 0) {
+            this.data.columns.push(column);
+        }
+    };
+
+    AutoflowTabularController.prototype.setObjects = function (domainObjects) {
+        this.releaseSubscriptions();
+        this.rows = {};
+        this.childObjects = domainObjects;
+        this.update();
+    };
+
+    AutoflowTabularController.prototype.setRows = function (rows) {
+        var changed = this.rowCount !== rows;
+        this.rowCount = rows;
+        if (changed) {
+            this.update();
+        }
+    };
+
+    AutoflowTabularController.prototype.activate = function () {
+        if (this.active) {
+            this.destroy();
```

Comment:
What exercises / requires this codepath?  It doesn't make sense to me that a component with a lifecycle can be reincarnated for a second round.